File: LEETCODE/trees/traversalTechniques/BredthFirstSearch./LeverOrderTraversal.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

import LEETCODE.trees.BinaryTree;


public class LeverOrderTraversal {
    // in this we wil queue data stricture 
    private static List<List<Integer>> lot(BinaryTree.Node root){
        
        List<List<Integer>> ans = new ArrayList<>();
        Queue<BinaryTree.Node> queue  = new LinkedList<>();

        // time: O(n) space : O(n) 
        queue.add(root);
        while (!queue.isEmpty()) {
            int size = queue.size();
            List<Integer> level = new ArrayList<>();
            for(int i =0; i < size; i++){
                BinaryTree.Node curr = queue.poll();
                level.add(curr.data);    
                if(curr.left !=  null){
                    queue.add(curr.left);
                }
                if(curr.right != null){
                    queue.add(curr.right);
                }
            }
            ans.add(level);
        }
        return ans;
    }
    public static void main(String[] args) {
        BinaryTree.Node root = BinaryTree.createTree();
        lot(root);
    }
}

File: LEETCODE/trees/traversalTechniques/BredthFirstSearch./LeverOrderTraversal.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode {
    int data;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int val) {
        this->data = val;
        this->left = nullptr;
        this->right = nullptr;
    }
};

vector<vector<int>> levelorder(TreeNode* root) {
    vector<vector<int>> ans;
    if(root == nullptr) {
        return ans;
    }
    queue<TreeNode*> q;
    q.push(root);
    while(!q.empty()) {
        int size = q.size();
        vector<int> level;
        for(int i = 0; i < size; i++) {
            auto curr = q.front();
            q.pop();
            // Add current node to this level
            level.push_back(curr->data);
            if(curr->left != nullptr) {
                q.push(curr->left);
            }
            if(curr->right != nullptr) {
                q.push(curr->right);
            }
        }
        ans.push_back(level);
    }

    return ans;
}